test/database.ts

```diff
@@ -2200,4 +2204,116 @@ describe('Database', () => {
+      assert.strictEqual(result, 'READY');
+    });
+  });
+
+  describe('getRestoreInfo', () => {
+    it('should get restore info from database metadata', async () => {
+      const restoreInfo = {sourceType: 'BACKUP'};
+      database.getMetadata = async () => [{restoreInfo}];
+      const result = await database.getRestoreInfo();
+      assert.deepStrictEqual(result, restoreInfo);
+    });
+  });
+
+  describe('getOperations', () => {
+    it('should create filter for querying the database', async () => {
+      const operations: IOperation[] = [{name: 'my-operation'}];
+
+      database.instance.getDatabaseOperations = async query => {
+        assert.strictEqual(query.filter, `name:${DATABASE_FORMATTED_NAME}`);
+        return [operations, {}];
+      };
+
+      const [results] = await database.getOperations();
+      assert.deepStrictEqual(results, operations);
+    });
+
+    it('should create filter for querying the database in combination with user supplied filter', async () => {
+      const operations: IOperation[] = [{name: 'my-operation'}];
+
+      database.instance.getDatabaseOperations = async query => {
+        assert.strictEqual(
+          query.filter,
+          `(name:${DATABASE_FORMATTED_NAME}) AND (someOtherAttribute: aValue)`
+        );
+        return [operations, {}];
+      };
+
+      const [results] = await database.getOperations({
+        filter: 'someOtherAttribute: aValue',
+      });
+      assert.deepStrictEqual(results, operations);
+    });
+  });
+
+  describe('restore', () => {
+    const BACKUP_NAME = 'backup-name';
+    const BACKUP_FORMATTED_NAME =
+      INSTANCE.formattedName_ + '/backups/' + BACKUP_NAME;
+
+    it('should make the correct request', async () => {
+      const QUERY = {};
+      const ORIGINAL_QUERY = extend({}, QUERY);
+      const expectedReqOpts = extend({}, QUERY, {
+        databaseId: NAME,
+        parent: INSTANCE.formattedName_,
+        backup: BACKUP_FORMATTED_NAME,
+      });
+
+      database.id = NAME;
+      database.request = config => {
+        assert.strictEqual(config.client, 'DatabaseAdminClient');
+        assert.strictEqual(config.method, 'restoreDatabase');
+        assert.deepStrictEqual(config.reqOpts, expectedReqOpts);
+
+        assert.notStrictEqual(config.reqOpts, QUERY);
+        assert.deepStrictEqual(QUERY, ORIGINAL_QUERY);
+      };
+
+      await database.restore(BACKUP_FORMATTED_NAME);
+    });
+
+    it('should accept a backup name', async () => {
+      const QUERY = {};
+      const ORIGINAL_QUERY = extend({}, QUERY);
+      const expectedReqOpts = extend({}, QUERY, {
+        databaseId: NAME,
+        parent: INSTANCE.formattedName_,
+        backup: BACKUP_FORMATTED_NAME,
+      });
+
+      database.id = NAME;
+      database.request = config => {
+        assert.deepStrictEqual(config.reqOpts, expectedReqOpts);
+      };
+
+      await database.restore(BACKUP_NAME);
+    });
+
+    describe('error', () => {
```

Comment:
Missing tests for the "success" path.

Reply:
Done. Also realized it was missing for Backup#create() as well so added that too.